pkg/controllers/rollout/controller_test.go

```diff
@@ -388,6 +395,80 @@ func TestPickBindingsToRoll(t *testing.T) {
 			tobeUpdatedBindings:        []int{0, 1, 2},
 			needRoll:                   true,
 		},
+		"test bound with failed to apply bindings when there is no max surge allowed": {
```

Comment:
failed to apply is actually mostly related to unavilable instead of surge

"test no binding will be updated when there is no max unavailable allowed": {
			allBindings: []*fleetv1beta1.ClusterResourceBinding{
				generateClusterResourceBinding(fleetv1beta1.BindingStateBound, "snapshot-1", cluster1),
				generateClusterResourceBinding(fleetv1beta1.BindingStateBound, "snapshot-1", cluster2),
				generateClusterResourceBinding(fleetv1beta1.BindingStateBound, "snapshot-1", cluster3),
				generateClusterResourceBinding(fleetv1beta1.BindingStateBound, "snapshot-1", cluster4),
				generateClusterResourceBinding(fleetv1beta1.BindingStateBound, "snapshot-1", cluster5),
			},
			latestResourceSnapshotName: "snapshot-2",
			crp:                        noMaxUnavailableCRP,
			tobeUpdatedBindings:        []int{},
			needRoll:                   true,
		},